plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -709,4 +741,99 @@ private List<Namespace> listNamespaces(ConnectorSession session, Optional<String
+                }
+                matchingTable = identifier;
+            }
+        }
+        return matchingTable == null ? TableIdentifier.of(remoteNamespace, tableIdentifier.name()) : matchingTable;
+    }
+
+    private TableIdentifier toRemoteView(ConnectorSession session, SchemaTableName schemaViewName)
+    {
+        TableIdentifier tableIdentifier = toIdentifier(schemaViewName);
+        if (caseInsensitiveNameMatching) {
+            try {
+                return remoteViewMappingCache.get(tableIdentifier, () -> findRemoteView(session, tableIdentifier));
+            }
+            catch (ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return tableIdentifier;
+    }
+
+    private TableIdentifier findRemoteView(ConnectorSession session, TableIdentifier tableIdentifier)
+    {
+        Namespace remoteNamespace = toRemoteNamespace(session, tableIdentifier.namespace());
+        List<TableIdentifier> tableIdentifiers = restSessionCatalog.listViews(convert(session), remoteNamespace);
+        TableIdentifier matchingView = null;
+        for (TableIdentifier identifier : tableIdentifiers) {
+            if (identifier.name().equalsIgnoreCase(tableIdentifier.name())) {
+                if (matchingView != null) {
+                    throw new TrinoException(NOT_SUPPORTED, "Duplicate view name is not supported with Iceberg REST catalog: " + Joiner.on(", ").join(matchingView.name(), identifier.name()));
+                }
+                matchingView = identifier;
+            }
+        }
+        return matchingView == null ? TableIdentifier.of(remoteNamespace, tableIdentifier.name()) : matchingView;
+    }
+
+    private Namespace toRemoteNamespace(ConnectorSession session, Namespace trinoNamespace)
+    {
+        if (caseInsensitiveNameMatching) {
+            try {
+                return remoteNamespaceMappingCache.get(trinoNamespace, () -> findRemoteNamespace(session, trinoNamespace));
+            }
+            catch (ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return trinoNamespace;
+    }
+
+    private Namespace findRemoteNamespace(ConnectorSession session, Namespace trinoNamespace)
+    {
+        List<Namespace> matchingRemoteNamespaces = getNamespaces(session, Namespace.empty()).stream()
+                .filter(ns -> toTrinoNamespace(ns).equals(trinoNamespace))
+                .collect(toImmutableList());
+        if (matchingRemoteNamespaces.size() > 1) {
+            throw new TrinoException(NOT_SUPPORTED, "Duplicate namespace name is not supported with Iceberg REST catalog");
+        }
+        return matchingRemoteNamespaces.isEmpty() ? trinoNamespace : matchingRemoteNamespaces.getFirst();
+    }
+
+    private List<Namespace> getNamespaces(ConnectorSession session, Namespace parentNamespace)
```

Comment:
I would rename to `listNamespaces`. I usually avoid `get` prefix when the method isn't lightweight operation.

Reply:
100% its just that at some point there was another method with the same name that I created and couldn't use list here. I renamed it now. Thanks